plugins/autostart/src/lib.rs

```diff
@@ -29,6 +27,16 @@ pub enum MacosLauncher {
     AppleScript,
 }
 
+#[derive(Debug, Clone)]
+pub enum PreferedName {
+    PackageInfoName,
+    PackageInfoCrateName,
+    ConfigIdentifier,
+    ConfigProductName,
+    Custom(String)
+}
```

Comment:
To be honest, I feel like we could just have 2 variants, a custom one, and a default one called `AppName`, and we can document that saying it's corresponding to the `productName` in the tauri config file and falls back to the cargo name

If a user wants to customize this further with thing like product name, they can use the builder in the `setup` hook, and they have the `App` available to them there

Reply:
> To be honest, I feel like we could just have 2 variants

I'd just use no enum. Have the current default be the default and let them specify a string if desired. Getting the desired value from tauri or cargo is imo easy enough to not have to deal with the logic ourselves.